python/source/flight.rst

```diff
@@ -0,0 +1,228 @@
+============
+Arrow Flight
+============
+
+Recipes related to leveraging Arrow Flight protocol
+
+.. contents::
+
+Using an Arrow Flight RPC server
+================================
+
+Suppose you want to implement a service that can store, send and receive
+parquet files using the Arrow Flight protocol, 
+``pyarrow`` provides an implementation framework in :mod:`pyarrow.flight` 
+and particularly through the :class:`pyarrow.flight.FlightServerBase` class.
+
+.. testcode::
+
+    import pathlib
+    import threading
+
+    import pyarrow as pa
+    import pyarrow.flight
+    import pyarrow.parquet
+
+
+    class FlightServer(pa.flight.FlightServerBase):
+
+        def __init__(self, location="grpc://0.0.0.0:8815", 
```

Comment:
As an alternative to hardcoding a port, you can bind to port 0, then print out `server.port` after creating the server instance (before calling serve() - confusingly, the server is started as soon as you instantiate the class)